query-service-impl/src/main/java/org/hypertrace/core/query/service/multivalue/MultiValueTransformation.java

```diff
@@ -0,0 +1,41 @@
+package org.hypertrace.core.query.service.multivalue;
+
+import io.reactivex.rxjava3.core.Observable;
+import io.reactivex.rxjava3.core.Single;
+import java.util.List;
+import javax.inject.Inject;
+import lombok.extern.slf4j.Slf4j;
+import org.hypertrace.core.query.service.AbstractQueryTransformation;
+import org.hypertrace.core.query.service.api.Expression;
+import org.hypertrace.core.query.service.api.Function;
+import org.slf4j.Logger;
+
+@Slf4j
+public class MultiValueTransformation extends AbstractQueryTransformation {
+  private final FunctionTransformation functionTransformation;
+
+  @Inject
+  MultiValueTransformation(FunctionTransformation functionTransformation) {
+    this.functionTransformation = functionTransformation;
+  }
+
+  @Override
+  protected Logger getLogger() {
+    return log;
+  }
+
+  @Override
```

Comment:
we're copy pasting the logic of the parent class in this function and the one below - instead, transform the function only and then send it back to super.